agent/consul/state/catalog.go

```diff
@@ -2884,6 +2887,38 @@ func (s *Store) ServiceTopology(
 			return 0, nil, fmt.Errorf("failed to fetch protocol for service %s: %v", sn.String(), err)
 		}
 
+		// Fetch connect endpoints for the target service in order to learn if its proxies are configured as
+		// transparent proxies.
+		if entMeta == nil {
+			entMeta = structs.DefaultEnterpriseMeta()
+		}
+		q := Query{Value: service, EnterpriseMeta: *entMeta}
+
+		idx, proxies, err := serviceNodesTxn(tx, ws, indexConnect, q)
+		if err != nil {
+			return 0, nil, fmt.Errorf("failed to fetch connect endpoints for service %s: %v", sn.String(), err)
+		}
+		if idx > maxIdx {
+			maxIdx = idx
+		}
+		if len(proxies) == 0 {
+			break
```

Comment:
took me a minute to realize you were using this to skip over the rest of the switch/case block